src/main/java/com/example/java/maven/game2048/UserInputProvider.java

```diff
@@ -3,14 +3,10 @@
 import java.util.Scanner;
 
 public class UserInputProvider {
-    public static int getIntFromUser(){
-        while (true){
-            try {
-                return new Scanner(System.in).nextInt();
-            } catch (Exception e) {
-                System.err.println("Dej mie inta");
-            }
-        }
-    }
 
+    public static String getMoveFromPlayer() {
+        System.out.println("Next move: ");
```

Comment:
use here `MessagePrinter` class

Reply:
I will not add more comments about it. But replace all `System.out.print`s in program with usage of `MessagePrinter`. `System.out.pintln` should be only in `MessagePrinter`

Reply:
Done.